### PR TITLE
Feature/merge develop

### PR DESCRIPTION
**Pull request by jrobinAV**

### CODE REVIEW

**Comment by jrobinAV** [comment]
@florian-vuillemot any review?

File: taipy/core/scenario/_scenario_manager.py [hunk 1]
```diff
@@ -198,23 +215,33 @@ def _compare(cls, *scenarios: Scenario, data_node_config_id: str = None):
         if len(scenarios) < 2:
             raise InsufficientScenarioToCompare
 
-        if not all([scenarios[0].config_id == scenario.config_id for scenario in scenarios]):
+        if not all(
+            [scenarios[0].config_id == scenario.config_id for scenario in scenarios]
```
Comment (florian-vuillemot) [inline taipy/core/scenario/_scenario_manager.py:219]:
```suggestion
            
            scenarios[0].config_id == scenario.config_id for scenario in scenarios
```

File: taipy/core/scenario/scenario.py [hunk 2]
```diff
@@ -203,28 +205,36 @@ def has_tag(self, tag: str) -> bool:
         return tag in self.tags
 
     def _remove_subscriber(self, callback: Callable):
-        self._subscribers.remove(callback)
+        elem = [x for x in self._subscribers if x.callback == callback]
+        if not elem:
+            raise ValueError
+        self._subscribers.remove(elem[0])
```
Comment (florian-vuillemot) [inline taipy/core/scenario/scenario.py:211]:
```suggestion
        elem = next(x for x in self._subscribers if x.callback == callback, None)
        if not elem:
            raise ValueError
        self._subscribers.remove(elem)
```

**Review by florian-vuillemot — APPROVED** [review]

File: taipy/core/scenario/scenario.py (hunk 2 above)
Comment (jrobinAV) [reply taipy/core/scenario/scenario.py:211]: to be done later

File: taipy/core/scenario/_scenario_manager.py (hunk 1 above)
Comment (jrobinAV) [reply taipy/core/scenario/_scenario_manager.py:219]: to be done later

**Pull request merged**